packages/react-component-library/src/components/RadioE/partials/StyledRadio.tsx

```diff
@@ -10,17 +10,33 @@ interface StyledRadioProps {
   $isChecked?: boolean
 }
 
-const RADIO_FOCUS_WIDTH = '0.1rem'
+const RADIO_ACTIVE_BORDER_WIDTH = '2px'
 
 const { spacing, fontSize, color, animation } = selectors
 
+const BackgroundColour = css<StyledRadioProps>`
+  ${({ $isDisabled }) =>
+    $isDisabled ? color('neutral', '000') : color('neutral', 'white')}
+`
+
+const CheckmarkActiveBorderColour = css<StyledRadioProps>`
+  ${({ $isDisabled }) =>
+    $isDisabled ? color('neutral', '200') : color('action', '500')}
+`
+
+const CheckmarkCheckedFillColour = css<StyledRadioProps>`
+  ${({ $isDisabled }) =>
+    $isDisabled ? color('neutral', '200') : color('action', '500')}
+`
+
 export const StyledRadio = styled.div<StyledRadioProps>`
```

Comment:
I know that this was already quite long but I do think it's a tricky maintenance point as it's effectively a function with nearly 100 lines and there are comments which sets alarm bells that it could be broken up.

Reply:
Yeah, definitely. It will need refactoring to fix #3056 so I'll add a comment to that issue